tests/generic_config_updater/test_incremental_qos.py

```diff
@@ -25,69 +38,185 @@ def ensure_dut_readiness(duthost):
     Args:
         duthost: DUT host object
     """
-    config_tmpfile = generate_tmpfile(duthost)
-    logger.info("config_tmpfile {}".format(config_tmpfile))
-    logger.info("Backing up config_db.json")
-    duthost.shell("sudo cp /etc/sonic/config_db.json {}".format(config_tmpfile))
     verify_orchagent_running_or_assert(duthost)
+    create_checkpoint(duthost)
 
     yield
  
-    verify_orchagent_running_or_assert(duthost)
-    logger.info("Restoring config_db.json")
-    duthost.shell("sudo cp {} /etc/sonic/config_db.json".format(config_tmpfile))
-    delete_tmpfile(duthost, config_tmpfile)
-    config_reload(duthost)
+    try:
+        verify_orchagent_running_or_assert(duthost)
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+    finally:
+        delete_checkpoint(duthost)
+
 
-    logger.info("TEARDOWN COMPLETED")
+def get_uplink_downlink_count(duthost, tbinfo):
+    """
+    Retrieves uplink and downlink count from DEVICE_NEIGHBOR_METADATA based on topology 
 
+    Args:
+        duthost: DUT host object
+        tbinfo: information about the running testbed
 
-def prepare_configdb_field(duthost, configdb_field, value):
+    Returns:
+        uplink count, downlink count
+
+    """
+    config_db_json = json.loads(duthost.shell("show runningconfig all")["stdout"])
+    device_neighbor_metadata = config_db_json["DEVICE_NEIGHBOR_METADATA"]
```

Comment:
>device_neighbor_metadata [](http://example.com/codeflow?start=4&length=24)

You may reuse `duthost.config_facts`
ref: [https://github.com/Azure/sonic-mgmt/blob/3a9720789b4cd4527865b0053795e98c14ccfd2c/tests/test_nbr_health.py#L81](https://github.com/Azure/sonic-mgmt/blob/3a9720789b4cd4527865b0053795e98c14ccfd2c/tests/test_nbr_health.py#L81)